pkg/auth/serviceaccount.go

```diff
@@ -57,3 +63,34 @@ func GetOIDCServiceAccountForResource(gvk schema.GroupVersionKind, objectMeta me
 		},
 	}
 }
+
+// EnsureOIDCServiceAccountExistsForResource makes sure the given resource has
+// an OIDC service account with an owner reference to the resource set.
+func EnsureOIDCServiceAccountExistsForResource(ctx context.Context, serviceAccountLister corev1listers.ServiceAccountLister, kubeclient kubernetes.Interface, gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta) error {
+	saName := GetOIDCServiceAccountNameForResource(gvk, objectMeta)
+	sa, err := serviceAccountLister.ServiceAccounts(objectMeta.Namespace).Get(saName)
+
+	// If the resource doesn't exist, we'll create it.
+	if apierrs.IsNotFound(err) {
+		logging.FromContext(ctx).Infow("Creating OIDC service account", zap.Error(err))
```

Comment:
I would even consider dropping this log line, if it fails we get the error, if it doesn't we know it's all good
```suggestion
		logging.FromContext(ctx).Debugw("Creating OIDC service account", zap.Error(err))
```